sharing/loop_through_rules.py

```diff
@@ -0,0 +1,229 @@
+        # if the particular ruleValue is a range type
+        bracket: str = ""
+        lower_limit: str = ""
+        upper_limit: str = ""
+
+        # Assign a variable for the ruleValue that is not a range type
+        filter_val: Union[str, float, int, None] = None
+
+        if isinstance(filter_by, str):
+
+            # Determine the type of bracket present in the ruleValue if it is a range type.
+            if (
+                re.search(pat3, filter_by) is not None
+                and re.search(pat4, filter_by) is not None
+            ):
+                bracket = "[]"
+                print("BRACKET PRESENT []")
+            if (
+                re.search(pat5, filter_by) is not None
+                and re.search(pat6, filter_by) is not None
+            ):
+                bracket = "()"
+                print("BRACKET PRESENT ()")
+            if (
+                re.search(pat5, filter_by) is not None
+                and re.search(pat4, filter_by) is not None
+            ):
+                bracket = "(]"
+                print("BRACKET PRESENT (]")
+            if (
+                re.search(pat3, filter_by) is not None
+                and re.search(pat6, filter_by) is not None
+            ):
+                bracket = "[)"
+                print("BRACKET PRESENT [)")
+
+        # Confirm if the ruleValue is range type by checking if bracket exist
+        # & create upper_limit and lower_limit value
+        if bracket and isinstance(filter_by, str):
+            if "," in filter_by:
+                filter_list = filter_by.split(",")
+                lower_limit = filter_list[0].strip()
+                lower_limit = lower_limit.replace(lower_limit[0], "")
+                lower_limit = lower_limit.strip()
+                lower_limit = lower_limit.lower()
+                upper_limit = filter_list[-1].strip()
+                upper_limit = upper_limit.replace(upper_limit[-1], "")
+                upper_limit = upper_limit.strip()
+                upper_limit = upper_limit.lower()
+            else:
+                filter_val = filter_by[0].strip()
+                filter_val = filter_val[1:-1]
+
+        # if the ruleValue is not a range type then store the single value
+        # to filter in variable 'filter_val'
+        else:
+            if isinstance(filter_by, str):
+                filter_val = filter_by.strip()
+            else:
+                filter_val = filter_by
+
+        # Create boolean variables that checks if the current filter value
+        # is a datetime or numeric or character value
+        rule_is_datetime: bool = False
+        rule_is_numeric: bool = False
+        rule_is_char: bool = False
+
+        # If the direction of the rule is row, remove rows of data based on rule
+        if rule["direction"] == "row":
+            column: bool = False
+            # if ruleValue is 'ALL', remove all columns and rows from dataframe
+            if filter_by == "ALL":
```

Comment:
Check comment on line 85 of loop_through_tables.py. Same for line 177 in this file.